### ISSUE TITLE
[runtimes] Support for trailing whitespace in atlas region names

### ISSUE THREAD

**Issue by richard-livingston** [EsotericSoftware/spine-runtimes#1622]
The atlas loaders trim all whitespace from either side of each line in the atlas, this causes problems when the region name deliberately contains trailing whitespace.

For example: 

```text
bird-s-0.png
size: 456, 456
format: RGBA8888
filter: Linear,Linear
repeat: none
shadow
  rotate: false
  xy: 0, 0
  size: 208, 288
  orig: 208, 288
  offset: 0, 0
  index: -1
body02 
  rotate: false
  xy: 0, 288
  size: 144, 144
  orig: 144, 144
  offset: 0, 0
  index: -1
```

In the above atlas example, there is region named **"bird02 "**, it has a space at the end of the name. Atlas loaders trim this whitespace and so we cannot find the image at runtime.

I propose that either:

- Lines are trimmed on the left only
- The Spine exporter does not allow trailing whitespace

Example loader with issue is: https://github.com/EsotericSoftware/spine-runtimes/blob/89d9474e4032d3d90a85cdf189c2423b281ff8c9/spine-ts/core/src/TextureAtlas.ts wherever the `trim()` function is called.

**Comment by badlogic** [comment]
It seems the texture packer in the Spine Editor does not trim trailing whitespace in image names. It should though, as the Spine Editor trims trailing whitespace everywhere else.

**Comment by NathanSweet** [comment]
Fixed in 3.9.
